### PR TITLE
testing/hexinject - new aport

### PR DESCRIPTION
**Pull request by archey**
Updated aport to fix build issues for hexinject

### CODE REVIEW

File: testing/hexinject/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Contributor: Arch3y <arch3y@riseup.net>
+# Maintainer: Fabian Affolter <fabian@affolter-engineering.ch>
+pkgname=hexinject
+pkgver=1.5
+pkgrel=0
+pkgdesc="A very versatile packet injector and sniffer"
+url="http://hexinject.sourceforge.net/"
+arch="all"
+license="BSD"
+depends=""
```
Comment (Shizmob) [inline testing/hexinject/APKBUILD]: If it's empty the line can safely be simply omitted.

File: testing/hexinject/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+makedepends="libpcap-dev"
+source="https://downloads.sourceforge.net/$pkgname/$pkgname-$pkgver.tar.gz gcc5-fixes.patch"
+
+_builddir="$srcdir/$pkgname"
```
Comment (Shizmob) [inline testing/hexinject/APKBUILD]: Please use `builddir` instead of `_builddir`.

File: testing/hexinject/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+pkgname=hexinject
+pkgver=1.5
+pkgrel=0
+pkgdesc="A very versatile packet injector and sniffer"
+url="http://hexinject.sourceforge.net/"
+arch="all"
+license="BSD"
+depends=""
+makedepends="libpcap-dev"
+source="https://downloads.sourceforge.net/$pkgname/$pkgname-$pkgver.tar.gz gcc5-fixes.patch"
```
Comment (Shizmob) [inline testing/hexinject/APKBUILD]: Please put all sources on their own lines:

```sh
sources="https://downloads.sourceforge.net/$pkgname/$pkgname-$pkgver.tar.gz
	gcc5-fixes.patch"
```

File: testing/hexinject/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+_builddir="$srcdir/$pkgname"
+
+prepare() {
```
Comment (Shizmob) [inline testing/hexinject/APKBUILD]: This is already done by the default `prepare()` and is thus not needed.

File: testing/hexinject/gcc5-fixes.patch [hunk 5]
```diff
@@ -0,0 +1,13 @@
+--- hexinject.orig/Makefile
++++ hexinject/Makefile
+@@ -1,7 +1,7 @@
+ all:
```
Comment (Shizmob) [inline testing/hexinject/gcc5-fixes.patch]: Might be cleaner to just add -std=gnu89 to the CFLAGS variable at the start.

File: testing/hexinject/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+package() {
+	cd "$_builddir"
+        install -Dm755 hexinject $pkgdir/usr/bini/hexinject
+	install -Dm755 prettypacket $pkgdir/usr/bin/prettypacket
+	install -Dm755 hex2raw $pkgdir/usr/bin//hex2raw
+}
+
+md5sums="26fbb396bed9c64de653e35ae78b6956  hexinject-1.5.tar.gz
```
Comment (Shizmob) [inline testing/hexinject/APKBUILD]: `md5sums=` and `sha256sums` aren't needed/used anymore.

File: testing/hexinject/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+build() {
+	cd "$_builddir"
+	make
+}
+
+package() {
+	cd "$_builddir"
+        install -Dm755 hexinject $pkgdir/usr/bini/hexinject
```
Comment (Shizmob) [inline testing/hexinject/APKBUILD]: 1) The indentation here seems off;
2) It'd be better to just use `cp` if the binaries already have the correct permissions;
3) Please quote `$pkgdir`;
4) Typo: `bini` -> `bin`;
5) Typo: `bin//` -> `bin/`;

**Comment by Shizmob** [comment]
Finally, I see that 1.6 is out -- may be better to package that directly?

**Comment by fabled** [comment]
Committed in c6304787520dad7650a8e16b4f87e083415561c2.

**Pull request closed without merging**